packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -158,4 +735,366 @@ library ConnextUtils {
     // Otherwise, swap to adopted asset
     return (pool.swapExact(_amount, _asset, adopted), adopted);
   }
+
+  /**
+   * @notice Performs some sanity checks for `xcall`
+   * @dev Need this to prevent stack too deep
+   */
+  function _xcallSanityChecks(xCallLibArgs calldata _args) private {
+    // ensure this is the right domain
+    if (_args.xCallArgs.params.originDomain != _args.domain) {
+      revert ConnextUtils__xcall_wrongDomain();
+    }
+
+    // ensure theres a recipient defined
+    if (_args.xCallArgs.params.to == address(0)) {
+      revert ConnextUtils__xcall_emptyTo();
+    }
+  }
+
+  /**
+   * @notice Processes an `xcall`
+   * @dev Need this to prevent stack too deep
+   */
+  function _processXcall(
```

Comment:
nit: can we call this `_xcallProcess` ? Would read better with `_xcallSanityChecks`... and would read better if we have to do this for other methods in the future (e.g. `_reconcileProcess`)

sorry